### DISCUSSION TITLE
Predicting road traffic

### DISCUSSION THREAD

**Discussion by Floriangr** [Universite-Gustave-Eiffel/NoiseModelling#303] [General]
Hi everyone,
I'm Florian, I'm working on providing noise pollution data for Germany (ruhecheck.de). I've been in touch with Pierre who mentioned it may be interesting for more to discuss here instead of via email. 

I'm currently building a model to calculate street noise pollution on the fly using Python. I'd like to get traffic volumes applicable to Germany and will build a model to predict that (I'm probably able to get a sufficiently large learning data set).
What do you think would be the best independent variables to predict traffic?
I would start with the following:
- Highway (from OSM)
- Max speed (from OSM)
- Betweenness Centrality of street segment (derived from OSM)
- Population density (from German census)

Anything I'm missing?

I'm looking forward to discussing with you!

Best,
Florian

**Comment by pierromond** [comment]
I am sure you could find a lot of information in this paper : https://www.researchgate.net/publication/346061860_A_proposal_for_producing_calculated_noise_mapping_defining_the_sound_power_levels_of_roads_by_street_stratification

"Another method is also proposed that estimates traffic flows. "
"he results demonstrated similar uncertainties in both methods and were in the range of expected average uncertainties compared to the results obtained by calculating a noise map with the measured experimental values. In particular, for the differences between calculations and measurements, in absolute values, the mean uncertainties were approximately 2 dBA in estimating different long-term noise indicators. "

**Reply by annezhangxue** [reply]
Hi Pierromond, in CNOSSOS (road traffic noise) model, is the single vehicle considered as point source or the traffic is handled as line source please? what about directivity factor Q,  set as 1 or 2 please?
![formula](https://user-images.githubusercontent.com/72456658/107833274-0798b300-6d93-11eb-8a93-3def29f0de82.png)
Thank you.

**Reply by pierromond** [reply]
Hi @annezhangxue! In fact, in the CNOSSOS, road traffic inputs linear sources BUT they are divided into several point sources before starting the attenuation calculation.
About the directivity factor, you can consult the [amendments published in December 2020](https://op.europa.eu/en/publication-detail/-/publication/fd974bbd-438e-11eb-b59f-01aa75ed71a1/language-en/format-PDF/source-search). This is well explained. If I remember correctly, the directivity factor is set as (omnidirectional) but a first reflection is taken into account in the emission coefficients.

Finally, feel free to open a new discussion topic if your question is not directly related to the initial question of the discussion flow. 

Cheers!

**Comment by pierromond** [comment]
This paper also can help you : https://www.researchgate.net/publication/345350483_Noise_Estimation_Using_Road_and_Urban_Features

**Reply by Floriangr** [reply]
Thanks a lot! Will have a look at the papers and update you on my results if you like.

**Reply by merghd** [reply]
Florian,

I've been developing a very similar thing in the UK, we've got traffic flow data published by the Government which can be used, and where missing data I just use the nearest similar road category data.

Unfortunately, in the UK the OSM buildings height data is not always comprehensive, so I get that from alternate sources - which aren't open so a small charge involved. But it's pretty straightforward to generate noise contours, or receiver values, automatically from OSM data. Especially in conjunction with Geoff Boeing's osmnx library, which helps a lot with network analysis of the traffic data.

I'd be happy to help out if you've got any questions.

**Reply by Floriangr** [reply]
Hi @merghd ,
thanks a lot for your reply! That's very interesting. I've also been working with osmnx which I find very helpful.
How about we have a chat on the topic next week? Maybe we can share some thoughts :)
Best regards,
Florian

**Reply by Floriangr** [reply]
Super! My email is florian.gruenwald@hec.edu. If you like just get in touch. @merghd

**Comment by pierromond** [comment]
I found that today... https://github.com/graphhopper/open-traffic-collection I am sure it can interest you !

**Reply by pierromond** [reply]
and this : https://github.com/piyushy1/OSMTrafficEstimation

**Reply by merghd** [reply]
I'd not seen the 2nd one - really interesting.

**Reply by Floriangr** [reply]
That's very interesting, thanks a lot for sharing, @pierromond !

**Comment by merghd** [comment]
FYI, just to give an idea of what I've been working on - this is a plot which is automatically generated from UK traffic data, building height data, and your software...
![image](https://user-images.githubusercontent.com/42804121/109634342-09a79380-7b41-11eb-98a4-cf83f8ab0002.png)

It's really quite powerful.

**Reply by merghd** [reply]
The magenta building is the one of interest.

**Reply by merghd** [reply]
All within a jupyter notebook for now, but class modules written and a flask framework nearly done

**Reply by pierromond** [reply]
really nice what kinf of 3d visualisation tool are you using ?

**Reply by merghd** [reply]
Either mapbox or OS - really easy to integrate geojson. Geopandas useful too and pydeck

**Comment by pierromond** [comment]
https://www.researchgate.net/publication/352905307_Predicting_traffic_noise_using_land-use_regression_-_a_scalable_approach

**Reply by Floriangr** [reply]
Interesting! Thank you for sharing, @pierromond !

**Reply by merghd** [reply]
Likewise, thanks - looks interesting
